src/shogun/base/SGObject.h

```diff
@@ -499,6 +501,16 @@ class CSGObject
 	 */
 	void set_seed(int32_t seed);
 
+	static void set_global_seed(uint32_t seed)
```

Comment:
there's no need for this here... since you seed is a global thing.
just in init as other things have
`set_global_seed` and `get_global_seed`